clickhouse_send_external_data.go

```diff
@@ -0,0 +1,35 @@
+package clickhouse
+
+import "github.com/ClickHouse/clickhouse-go/lib/data"
+
+func (ch *clickhouse) sendExternalTables(externalTables []ExternalTable) error {
+	ch.logf("[send external tables] count %d", len(externalTables))
+	if externalTables == nil || len(externalTables) == 0 {
+		return nil
+	}
+	block := &data.Block{}
+	sentTables := make(map[string]bool, 0)
+	for _, externalTable := range externalTables {
+		if _, ok := sentTables[externalTable.Name]; ok {
+			continue
+		}
+		ch.logf("[send external table] name %s", externalTable.Name)
+		sentTables[externalTable.Name] = true
+		block.Columns = externalTable.Columns
+		block.NumColumns = uint64(len(externalTable.Columns))
+		for _, row := range externalTable.Values {
```

Comment:
I wonder if it'd make sense to expose `*Block` as an option for `ExternalTable` since that'd mean we don't end up moving the data around twice